src/app/components/wrapper-components/part-creation/part-creation.component.ts

```diff
@@ -214,7 +223,7 @@ export class PartCreationComponent implements OnInit, OnDestroy {
         this.viewInfo.totalPartDuration = this.viewInfo.totalPartDuration || joiner.totalPartDuration;
         let opponent: string | undefined = this.viewInfo.chosenOpponent;
         if (opponent) {
-            if (joiner.candidates.indexOf(opponent) === -1) {
+            if (joiner.candidates.some((minimalUser: MinimalUser) => minimalUser.name === opponent) === false) {
```

Comment:
idem + à moultes endroits, si tu te sens chaud de faire le switch username -> userid dans la logique, ça serait idéal

Reply:
Pour l'instant dans la viewInfo il n'y a pas de id qu'on envoi (c'qui peut éventuellement être vu comme une sécurité) mais à part dans **getUserFromName(userName): MinimalUser**, dans partCreation tous les === comparent des id entre eux ! aucuns des name !

Reply:
ook nice! Que l'absence d'id dans la viewInfo soit une sécurité, j'en suis pas trop convaincu par contre. Tu peux résoudre if you want